<!-- Run the aruco_detect node -->
<launch>
  <!-- namespace for camera input -->
  <arg name="camera" default="/raspicam_node"/>
  <arg name="image" default="image"/>
  <arg name="transport" default="compressed"/>
  <arg name="fiducial_len" default="0.14"/>
  <arg name="dictionary" default="7"/>
  <arg name="do_pose_estimation" default="true"/>
  <!-- If vis_msgs set to true, pose estimation will be published with ROS standard vision_msgs -->
  <arg name="vis_msgs" default="false"/>
  <arg name="ignore_fiducials" default="" />
  <arg name="fiducial_len_override" default="" />
  <arg name="verbose" default="false"/>

  <node pkg="aruco_detect" name="aruco_detect"
    type="aruco_detect" output="screen" respawn="false">
    <param name="image_transport" value="$(arg transport)"/>
    <param name="publish_images" value="true" />
    <param name="fiducial_len" value="$(arg fiducial_len)"/>
    <param name="dictionary" value="$(arg dictionary)"/>
    <param name="do_pose_estimation" value="$(arg do_pose_estimation)"/>
    <param name="vis_msgs" value="$(arg vis_msgs)"/>
    <param name="ignore_fiducials" value="$(arg ignore_fiducials)"/>
    <param name="fiducial_len_override" value="$(arg fiducial_len_override)"/>
    <param name="verbose" value="$(arg verbose)"/>
    <remap from="camera/compressed" 
        to="$(arg camera)/$(arg image)/$(arg transport)"/>
    <remap from="camera_info" to="$(arg camera)/camera_info"/>
  </node>
</launch>
